frontend/src/components/form/SearchUsers.vue

```diff
@@ -74,11 +73,17 @@ const onInput = async (event: any) => {
       if( input.length >= 3 ) {
         await userStore.fetchUsers({ idp: selectedIDP.value.idp, search: input });
       }
+      else {
+        userStore.clearSearch();
+      }
```

Comment:
1) Is there a way we can determine what kind of type we are getting for the event callback? We'd want to avoid using the any type wherever possible. If no such type exists in the vue types, then we should be more specific with the declaration.

2) While the nested boolean logic here makes sense due to necessity, it is getting dangerously close to a high cognitive complexity level. I propose we do the following to attempt to flatten out its structure:

```ts
const onInput = async (event: { target: { value: string } }) => {
  const input: string = event.target.value;
  if ( selectedIDP.value?.idp ) {
    if ( selectedIDP.value?.searchable && input.length >= 3 ) {
      await userStore.fetchUsers({ idp: selectedIDP.value.idp, search: input });
    } else if ( input.match( Regex.EMAIL ) ) {
      await userStore.fetchUsers({ idp: selectedIDP.value.idp, email: input });
    } else {
      userStore.clearSearch();
    }
  }
};
```
3) Even with the above, do we actually need the async/await on this event? We're not explicitly needing to do an operation right after that requires the blocking aspect of the await, so we might be able to get away with making this a synchronous callback function instead, and let the asynchronous userStore actions complete at their leisure.